### DISCUSSION TITLE
Modules

### DISCUSSION THREAD

**Discussion by Pasqualino1503** [netbox-community/netbox#10657] [General]
I been trying to install bulk modules  but  keep getting an error:

<class 'dcim.models.devices.Module.module_bay.RelatedObjectDoesNotExist'>

Module has no module_bay.

Python version: 3.8.10
NetBox version: 3.3.5

I did double checked if the  module_bays exist  which they do.

Wondering what can it be

**Comment by candlerb** [comment]
Please can you explain exactly what steps you're doing when you are "trying to install bulk modules"?

* Which page you're going to / which URL you see in the URL bar
* What you're clicking on
* What data you're entering
* etc

Also, see if you can provide a backtrace. The easiest way to do this is to set EMAIL and ADMINS in configuration.py; whenever an exception occurs, the full backtrace is E-mailed out.

**Reply by Pasqualino1503** [reply]
Thanks  for the reply. I'm using the demo.netbox.dev
Not sure  if I need to send  the  devices, modules, and also the install modules.

**Reply by candlerb** [reply]
Yes, you need to provide the complete set of steps to reproduce on a fresh installation.

At the moment it's not even clear if you're using CSV upload, or something else.  And we don't know what device types you've created, and what components they have.

**Reply by Pasqualino1503** [reply]
so here it goes
[404.04 FppA1 Modules5.xlsx](https://github.com/netbox-community/netbox/files/9799931/404.04.FppA1.Modules5.xlsx)
[NetBox Device Import2.xlsx](https://github.com/netbox-community/netbox/files/9799945/NetBox.Device.Import2.xlsx)
[netbox_CCH 044 12V.csv](https://github.com/netbox-community/netbox/files/9799949/netbox_CCH.044.12V.csv)
[netbox_Edge_4U.csv](https://github.com/netbox-community/netbox/files/9799954/netbox_Edge_4U.csv)
[netbox_Edge8_2U.csv](https://github.com/netbox-community/netbox/files/9799955/netbox_Edge8_2U.csv)
[netbox_Edge8_4U.csv](https://github.com/netbox-community/netbox/files/9799957/netbox_Edge8_4U.csv)
[Netbox-Module_type Corning CP32-V3.xlsx](https://github.com/netbox-community/netbox/files/9799958/Netbox-Module_type.Corning.CP32-V3.xlsx)
[Netbox-Module_type Corning CP48-E3.xlsx](https://github.com/netbox-community/netbox/files/9799960/Netbox-Module_type.Corning.CP48-E3.xlsx)
[Netbox-Module_type Corning ECM8-UM08-04-E8G-ULL.xlsx](https://github.com/netbox-community/netbox/files/9799961/Netbox-Module_type.Corning.ECM8-UM08-04-E8G-ULL.xlsx)
[Netbox-Module_type Corning ECM8-UM08-05-E6Q-ULL.xlsx](https://github.com/netbox-community/netbox/files/9799962/Netbox-Module_type.Corning.ECM8-UM08-05-E6Q-ULL.xlsx)
[Netbox-Module_type Corning ECM-UM-12-04-89G.xlsx](https://github.com/netbox-community/netbox/files/9799963/Netbox-Module_type.Corning.ECM-UM-12-04-89G.xlsx)
[Netbox-Module_type Corning ECM-UM-12-05-93T.xlsx](https://github.com/netbox-community/netbox/files/9799967/Netbox-Module_type.Corning.ECM-UM-12-05-93T.xlsx)

**Reply by Pasqualino1503** [reply]
these  are all the files ,let me know if i'm doing something wrong please

**Reply by candlerb** [reply]
That's not a series of steps to reproduce the problem.
* How exactly are those .xlsx (Excel) sheets used? Netbox doesn't work with Excel files AFAIK.
* I looked at one of the "CSV" files - `netbox_Edge_4U.csv` - and it isn't even CSV, it's YAML.

**Reply by Pasqualino1503** [reply]
this  is not the way

**Reply by Pasqualino1503** [reply]
I created them  and then pasted it to

**Reply by Pasqualino1503** [reply]
if  you go to the demo the  devices and modules  all seem to be there

**Reply by Pasqualino1503** [reply]
I'm really sorry, I watched  a few videos, this is what  I understood.

**Reply by candlerb** [reply]
I drop out here: I'm afraid I don't have time to play guessing games. Good luck with resolving your problem.

**Reply by Pasqualino1503** [reply]
I'm completely lost  here  , I was asked  to transfer my old project  to Netbox  with no help
